server/issue.go

```diff
@@ -30,7 +30,7 @@ func makePost(userId, channelId, message string) *model.Post {
 	}
 }
 
-func respondErr(plugin *Plugin, status int, e error, mmuserID, jiraBotID, channelID, message string) (int, error) {
+func respondEphemeralErr(plugin *Plugin, status int, e error, mmuserID, jiraBotID, channelID, message string) (int, error) {
```

Comment:
@mickmister, @levb - Any reason not to convert the function a method and the plugin as the receiver?

Reply:
If similar functions are using the same argument structure, I would keep it as is. Otherwise yes that seems better.

Also `issue.go` seems like a weird place to house this function. Is there a spot where similar functions are located?

Reply:
I moved the renamed function `responseEphemeralErr()` into `server/utils.go`.

There is also a `server/utils/utils.go` file.  Perhaps they could be combined in another ticket?

I'd also recommend another ticket to refactor the function to accept a newly created struct as an input.  Each call to `responseEphemeralErr()` is pretty length.

Reply:
> There is also a server/utils/utils.go file. Perhaps they could be combined in another ticket?

Sounds good :+1:

> I'd also recommend another ticket to refactor the function to accept a newly created struct as an input. Each call to responseEphemeralErr() is pretty length.

The point of using these helper functions is to avoid making a struct from the caller side, e.g. `Post`, and instead have typed/named function arguments that guide you to correctness. The call to `responseEphemeralErr` is lengthy because of how many inputs it requires, and you can't really change that, other than removing the `jiraBotID` param.

Maybe instead the code that is using this function can be restructured slightly so the function is not needed, especially since the function is not used much.

Reply:
I actually have a refactored version that I made before reverting.  `repsondEphemeralErr()` call can be replaced by the following:

This would remove the function altogether and I think makes sense from a readability standpoint.  What do you think?

```go
	if !ok {
		msg = "No transition option was found in context data"
		_ = plugin.API.SendEphemeralPost(mattermostUserId, makePost(jiraBotID, channelID, msg))
		return http.StatusInternalServerError, errors.New(msg)
	}
```

Reply:
LGTM :+1: